samtranslator/model/api/api_generator.py

```diff
@@ -565,6 +574,15 @@ def _construct_api_domain(  # noqa: too-many-branches
             logical_id = "RecordSetGroup" + logical_id_suffix
 
             record_set_group = route53_record_set_groups.get(logical_id)
+
+            if route53.get("SeparateRecordSets"):
```

Comment:
`SeparateRecordSets` cannot be intrinsic right? Otherwise this might be an intrinsic resolving to `False`

Reply:
Correct, this might wrongly enable it. Need to validate type as it's a nested value. :/

Reply:
Added type validation and intrinsic check.